tests/clients/test_alerts.py

```diff
@@ -92,3 +138,58 @@ def test_alerts_client_calls_get_aggregate_data_with_expected_value_and_param(
         alert_client = AlertsClient(mock_alerts_service, mock_alert_rules_service)
         alert_client.get_aggregate_data("alert-id")
         mock_alerts_service.get_aggregate_data.assert_called_once_with("alert-id")
+
+    def test_alerts_client_get_all_alert_details_calls_get_details_for_each_page(
+        self, mock_alerts_service_with_pages, mock_alert_rules_service
+    ):
+        mock_alerts_service = mock_alerts_service_with_pages(ascending=True)
+        alert_client = AlertsClient(mock_alerts_service, mock_alert_rules_service)
+        query = AlertQuery()
+        list(alert_client.get_all_alert_details(query))
```

Comment:
Ahh nice I did not know about this Python idiom for retrieving all the results of a generator! Seems very handy for debugging.